src/wazuh_qa_framework/system/wazuh_handler.py

```diff
@@ -688,34 +690,43 @@ def restart_managers(self, manager_list, parallel=True):
                 self.restart_manager(manager)
         self.logger.info(f"Managers restarted successfully: {manager_list}")
 
-    def stop_agent(self, host):
-        """Stop agent
+    def stop_agent(self, host, gracefully=True):
+        """Stop agent.
 
         Args:
-            host (str): Hostname
+            host (str): Hostname.
+            gracefully (bool): Stop gracefully. Defaults to True.
         """
-        self.logger.debug(f'Stopping agent {host}')
+        type = 'gracefully' if gracefully else 'ungracefully'
+        self.logger.debug(f'Stopping {type} {host}')
         service_name = WAZUH_AGENT_WINDOWS_SERVICE_NAME if self.is_windows(host) else 'wazuh-agent'
         if self.is_agent(host):
-            self.control_service(host, service_name, 'stopped')
-            self.logger.debug(f"Agent {host} stopped successfully")
+            if not gracefully:
+                process = 'wazuh-agent.exe' if self.is_windows(host) else 'wazuh-agent'
+                self.stop_ungracefully_process(host, process)
+
+            else:
+                self.control_service(host, service_name, 'stopped')
+            self.logger.info(f'Agent {host} stopped {type}')
+
         else:
             raise ValueError(f"Host {host} is not an agent")
 
-    def stop_agents(self, agent_list=None, parallel=True):
-        """Stop agents
+    def stop_agents(self, agent_list=None, parallel=True, gracefully=True):
+        """Stop agents.
 
         Args:
-            agent_list(list, optional): Agents list. Defaults to None
+            agent_list (list, optional): Agents list. Defaults to None.
             parallel (bool, optional): Parallel execution. Defaults to True.
+            gracefully (bool): Stop gracefully. Defaults to True.
         """
-        self.logger.info(f"Stopping agents: {agent_list}")
         if parallel:
-            self.pool.starmap(self.stop_agent, agent_list)
+            stop_agent_with_type = partial(self.stop_agent, gracefully=gracefully)
+            self.pool.map(stop_agent_with_type, agent_list)
```

Comment:
I suggest to use startmap for consistency with the rest of the methods
```suggestion
            self.pool.starmap(stop_agent_with_type, [(agent, gracefully) for agent in agent_list])
```

Reply:
Done in https://github.com/wazuh/qa-system-framework/pull/72/commits/1ebf89058de3b4a0eae192c0ede83285e0480140